compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangAnonymousModelHelper.java

```diff
@@ -97,6 +101,34 @@ public String getNextAnonymousTypeKey(PackageID packageID) {
         return ANON_TYPE + UNDERSCORE + nextValue;
     }
 
+    public String getNextAnonymousTypeKey(PackageID packageID, Stack<String> suffixes) {
```

Comment:
Shouldn't we make `getNextAnonymousTypeKey(PackageID package)` method a private one? We still use that API in some places.

Reply:
Since there are places where we don't need to create identifiers that are not needed to identify uniquely, I think we need this API.